src/gcp/cloudscheduler.ts

```diff
@@ -12,38 +14,116 @@ export interface Schedule {
+*/
+export function deleteJob(name: string): Promise<any> {
   return api.request("DELETE", `/${VERSION}/${name}`, {
     auth: true,
     origin: api.cloudschedulerOrigin,
   });
 }
 
-export async function createOrReplaceJob(schedule: Schedule): Promise<void> {
-  const jobName = `${schedule.name.split("/")[5]}`;
+/*
+* Gets a cloudScheduler job with the given name.
+* If no job with that name exists, this will return a 404.
+* @param name: The name of the job to get.
+*/
+export function getJob(name: string): Promise<any> {
+  return api.request("GET", `/${VERSION}/${name}`, {
+    auth: true,
+    origin: api.cloudschedulerOrigin,
+  });
+}
+
+/*
+* Updates a cloudScheduler job.
+* Returns a 404 if no job with that name exists.
+* @param job: A job to update. Note that name cannot be updated.
+*/
+export function updateJob(job: Job): Promise<any> {
+  return api.request("PATCH", `/${VERSION}/${job.name}`, {
+    auth: true,
+    origin: api.cloudschedulerOrigin,
+    data: Object.assign({ timeZone: DEFAULT_TIME_ZONE }, job),
+  });
+}
+
+/*
+* Checks for a existing job with the given name.
+* If none is found, it creates a new job.
+* If one is found, and it is identical to the job parameter, it does nothing.
+* Otherwise, if one is found and it is different from the job param, it updates the job.
+* @param job: A job to check for and create, replace, or leave as appropriate.
+* @throws { FirebaseError } if an error respnse other than 404 is recieved on the GET call.
+*/
+export async function createOrReplaceJob(job: Job): Promise<any> {
+  const jobName = `${job.name.split("/")[5]}`;
   try {
-    await createJob(schedule);
-    logLabeledSuccess("functions", `created scheduler job ${jobName}`);
+    const existingJob = await getJob(job.name);
+    if (!job.timeZone) {
+      // We set this here to avoid recreating schedules that use the default timeZone
+      job.timeZone = DEFAULT_TIME_ZONE;
+    }
+
+    if (isIdentical(existingJob.body, job)) {
+      logLabeledBullet("functions", `scheduler job ${jobName} is up to date, no changes required`);
+      return;
+    } else {
+      const updatedJob = await updateJob(job);
+      logLabeledBullet("functions", `updated scheduler job ${jobName}`);
+      return updatedJob;
+    }
   } catch (e) {
-    if (e.context.response.statusCode !== 409) {
+    // If the error status is 404, no job exists, so we can create one
+    // If it is anything else, we should error out
+    if (e && e.context && e.context.response && e.context.response.statusCode !== 404) {
       throw e;
     }
-    logLabeledBullet("functions", `re-creating scheduler job ${jobName}`);
-    await deleteJob(schedule.name);
-    return createJob(schedule);
+    const newJob = await createJob(job);
+    logLabeledSuccess("functions", `created scheduler job ${jobName}`);
+    return newJob;
   }
 }
+
+/*
+* Check if two jobs are functionally equivalent.
+* @param existingJob: a job to compare.
```

Comment:
it's ok to exclude the @param comment if the variable name is descriptive enough.
I think in a case like this it's clear we're comparing the existingJob and newJob so I'd be ok omitting it, but up to you.